File: 15_simple_cache_system/java/SimpleCache.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class SimpleCache {
    public static void main(String[] args) {

        Map<String, String> cache = new HashMap<>();
        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.println("\n1. Add to Cache");
            System.out.println("2. Get from Cache");
            System.out.println("3. View Cache");
            System.out.println("4. Exit");
            System.out.print("Choice: ");

            String choice = sc.nextLine();

            switch (choice) {
                case "1":
                    System.out.print("Key: ");
                    String key = sc.nextLine();
                    System.out.print("Value: ");
                    String value = sc.nextLine();
                    cache.put(key, value);
                    System.out.println("Stored in cache.");
                    break;

                case "2":
                    System.out.print("Key: ");
                    key = sc.nextLine();
                    System.out.println("Value: " + cache.getOrDefault(key, "Key not found"));
                    break;

                case "3":
                    System.out.println("Cache Contents:");
                    for (Map.Entry<String, String> entry : cache.entrySet()) {
                        System.out.println(entry.getKey() + " : " + entry.getValue());
                    }
                    break;

                case "4":
                    sc.close();
                    return;

                default:
                    System.out.println("Invalid choice.");
            }
        }
    }
}

File: 15_simple_cache_system/python/simple_cache.py
cache = {}

while True:
    print("\n1. Add to Cache")
    print("2. Get from Cache")
    print("3. View Cache")
    print("4. Exit")

    choice = input("Choice: ")

    if choice == "1":
        key = input("Key: ")
        value = input("Value: ")
        cache[key] = value
        print("Stored in cache.")

    elif choice == "2":
        key = input("Key: ")
        print("Value:", cache.get(key, "Key not found"))

    elif choice == "3":
        print("Cache Contents:")
        for k, v in cache.items():
            print(k, ":", v)

    elif choice == "4":
        break

    else:
        print("Invalid choice.")
